src/probnum/quad/_integration_measures.py

```diff
@@ -0,0 +1,267 @@
+"""Contains integration measures."""
+
+import abc
+from typing import Optional, Tuple, Union
+
+import numpy as np
+import scipy.stats
+
+from probnum.randvars import Normal
+from probnum.type import FloatArgType, IntArgType, RandomStateArgType
+
+
+class IntegrationMeasure(abc.ABC):
+    """An abstract class for a measure against which a target function is integrated.
+
+    Child classes implement specific integration measures and, if available, make use
+    of random variables for sampling and evaluation of the density function.
+
+    Parameters
+    ----------
+    dim :
+        Dimension of the integration domain
+    domain :
+        Tuple which contains two arrays which define the start and end points,
+        respectively, of the rectangular integration domain.
+    random_state :
+        Random state of the random variable corresponding to the integration measure.
+    """
+
+    def __init__(
+        self,
+        dim: IntArgType,
+        domain: Tuple[Union[np.ndarray, FloatArgType], Union[np.ndarray, FloatArgType]],
+        random_state: Optional[RandomStateArgType] = None,
+    ) -> None:
+
+        self._set_dimension_domain(dim, domain)
+        self.random_state = random_state
+
+    def __call__(self, points: Union[float, np.floating, np.ndarray]) -> np.ndarray:
+        """Evaluate the density function of the integration measure.
+
+        Parameters
+        ----------
+        points :
+            Input locations (n_points,) or (n_points, dim)
```

Comment:
Shape hints in ProbNum are specified as `*shape=(n, n)* -- Input locations.` See also the developer guide on documenting code: https://probnum.readthedocs.io/en/latest/development/adding_to_the_api_documentation.html